pkg/datapath/linux/routing/routing.go

```diff
@@ -235,17 +309,24 @@ func Delete(ip netip.Addr, compat bool) error {
 		"ip": ipWithMask.String(),
 	})
 
-	// Ingress rules
 	ingress := route.Rule{
 		Priority: linux_defaults.RulePriorityIngress,
 		To:       ipWithMask,
 		Table:    route.MainTable,
 	}
-	if err := deleteRule(ingress); err != nil {
-		return fmt.Errorf("unable to delete ingress rule from main table with ip %s: %w", ipWithMask.String(), err)
-	}
 
-	scopedLog.WithField("rule", ingress).Debug("Deleted ingress rule")
+	// Ingress rules
+	if ip.Is4() {
+		if err := deleteRuleIPv4(ingress); err != nil {
```

Comment:
nit: Similar to above (with replaceRule), consider introducing a local variable called `deleteRuleFn` to which you assign `deleteRuleIPv4` or `deleteRuleIPv6` and then call it here unconditionally.